app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -136,6 +136,10 @@ def backend_service_type(repository)
         end
       end
 
+      def pulp3_enabled?
+        !self.capabilities('Pulp3').nil?
```

Comment:
This isn't the best way to test, i think has_feature?('Pulp3')  is better.  What you're doing is fetching the capabiltiies and checking if its nil, which is kind of a side effect of it not having the pulp3 feature